### ISSUE TITLE
Is it possible to speed up initial import?

### ISSUE THREAD

**Issue by kmpm** [eyeonus/Trade-Dangerous#33]
The initial import of StationItems take forever.... Could it be speed up?

When I ran the initial update from eddb I got
```
...
NOTE: Processing market data from listings.csv: Start time = 2019-01-03 15:37:54.424876
NOTE: Finished processing market data. End time = 2019-01-03 15:58:23.628491
...
```
Processing market data took 21 minutes when starting from "empty" data folder.

Since the command line sqlite3 tool have support for importing csv files I made some tests, see attached script [import-help.sql.txt](https://github.com/eyeonus/Trade-Dangerous/files/2724338/import-help.sql.txt) .
```
2019-01-03 15:35:06|Importing eddb/listings.csv
2019-01-03 15:35:23|Generating Indexes
2019-01-03 15:35:29|Fixing brackets
2019-01-03 15:35:31|Creating 'simplified' StationItem table
2019-01-03 15:35:31|Insert or replace StationItems
2019-01-03 15:35:54|Done!
``` 
48 seconds......!
I ran this by doing `rm test.db & sqlite3 test.db ".read import-help.sql.txt"` in the Trade-Dangerouse folder after listings etc had been downloaded.

After hacking the code in eddblink_plug and importing using the sqlite3 command+script for `listings.csv` but not creating a modified `StationItem` table I got.
```
...
NOTE: Processing market data from listings.csv: Start time = 2019-01-03 17:23:15.991480
2019-01-03 16:23:16|Importing eddb/listings.csv
2019-01-03 16:23:37|Generating Indexes
2019-01-03 16:23:46|Fixing brackets
2019-01-03 16:23:50|Insert or replace StationItems
2019-01-03 16:27:32|Done!
NOTE: Finished processing market data. End time = 2019-01-03 17:27:32.298064
...
```
Worse with all the foreign key  checking and existing related data but ~5 minutes is way better than 21.
(The 1 hour difference in timestamps depends on me being UTC+1)

Is there something I'm missing?

OS: Windows 10, 1809
CPU: i7-8650U
Python 3.5.4 (v3.5.4:3f56838, Aug  8 2017, 02:17:05) [MSC v.1900 64 bit (AMD64)] on win32

**Comment by eyeonus** [comment]
The script only works on listings.csv, whereas the plugin also needs to import listings-live.csv

The script doesn't check to make sure that the listings it's checking to import is newer than the listing it would replace, it just inserts or replaces everything. If the data in the DB is newer than the data from the import file, it should be skipping that item, and it isn't.

What should be done, and hasn't mainly due to lack of time, is to not perform each transaction individually, which is what is currently happening, but to push all the transactions into a list for an executemany call.

The biggest slowdown with the current method is that each transaction is done individually, so the DB gets opened, written, closed, open, written, closed, for every entry in the import file.

It's the constant opening and closing that is the biggest slowdown.

I originally wrote the plugin with literally 0 knowledge of SQLite3 beyond what I was Googling to get the thing to work, I just haven't gone back to redo the importing in a less stupid way.

TBH, every one of the import methods in the plugin really needs to be improved in the same manner.

**Comment by aadler** [comment]
Over my head, but there are some examples of executemany in [cache.py](https://github.com/eyeonus/Trade-Dangerous/blob/c0fd39e7f325950182fd9e9505277c36fa9e16da/cache.py#L813) if that helps.

**Comment by aadler** [comment]
This should probably be closed now due to 48686db3ddfbf6e5dffc70087565c63d6d3a01c4, no?
